### DISCUSSION TITLE
Automerge on approval, without enforcing approvals

### DISCUSSION THREAD

**Discussion by gjoseph** [renovatebot/renovate#12368] [Request Help]
Is there (should there be) a way to tell renovate to auto-merge PRs only if/when they're approved, without configuring my GitHub repo to enforce approval? I'd like to keep an eye on major or even minor updates before they get merged, but I want patches to just automerge; I can't tell that to GitHub, so my repo is currently configured to not require PRs and approvals, but I also don't know how to tell renovate to "please wait for approval anyway".

One reason I don't want to enforce approvals at the GitHub level is ... that would prevent me from merging my own PRs, unless I use for "bypass cause I'm admin", which kinda defeats the purpose and adds more clicks than I want to.

My current workaround - use `pr-comment` and a predefined comment (which I'm not yet sure actually works, because for no good reason I wanted to have an emoji in that comment, and that may well be the reason why I haven't seen merges yet :D)

**Comment by viceice** [comment]
dashboard approval for renovate updates 👉 https://docs.renovatebot.com/key-concepts/dashboard/#require-approval-for-major-updates

automerge 👉 https://docs.renovatebot.com/key-concepts/automerge/

**Reply by rarkins** [reply]
What about this workflow?
1. Configure Renovate to automerge patch only
2. For minor or major, instead of "approving" the PR, you instead enable GitHub's native automerge and let GitHub do it

**Reply by gjoseph** [reply]
I briefly looked at the dashboard-approval, but I assumed this wouldn't show me the diff or the changelog.

Github's automerge... duh! Yes, that seems obvious now that you mention it :)

**Reply by gjoseph** [reply]
ha! I also realised I completely misunderstand the `automergeComment` -- I thought this was "if a user leaves this comment, then renovate will auto-merge", whereas it's "renovate will leave this comment if it's not already there, so that _some other tool_ will manage the merge" 🤦
